drivers/s3.go

```diff
@@ -372,6 +373,22 @@ func (os *s3Session) saveDataPut(ctx context.Context, name string, data io.Reade
 	return url, nil
 }
 
+func (os *s3Session) DeleteFile(ctx context.Context, name string) error {
+	if os.s3svc == nil {
+		return errors.New("delete not supported for non full api")
+	}
+	params := &s3.DeleteObjectInput{
+		Bucket: aws.String(os.bucket),
+		Key:    aws.String(name),
```

Comment:
Same for joining session key

Reply:
Doesn't this already happen on the lines below?

Reply:
It has a little different logic, actually only using the key if the provided name is empty. I think we should copy this logic instead: https://github.com/livepeer/go-tools/blob/923fecd7eec57802b8037ee014026430e6b2e03a/drivers/s3.go#L303-L306

Reply:
Gotcha, have copied that logic now